ckan/views/admin.py

```diff
@@ -156,6 +157,34 @@ def __init__(self):
             }
         }
 
+    def _get_deleted_datasets(self):
+        if config.get_value('ckan.search.remove_deleted_packages'):
+            return self._get_deleted_datasets_from_db()
+        else:
+            return self._get_deleted_datasets_from_search_index()
+
+    def _get_deleted_datasets_from_db(self):
+        return model.Session.query(
+            model.Package
+        ).filter_by(
+            state=model.State.DELETED
+        )
+
+    def _get_deleted_datasets_from_search_index(self):
+        query = search.query_for(model.Package)
+        search_params = {
+            'fq': '+state:deleted',
+            'df': 'text',
+            'fl': 'id name title',
```

Comment:
You can add `dataset_type` to `fl` to get the `result['type']`. It's probably better anyway than hardcoding it to support purging other datasets types

Reply:
Fixed [here](https://github.com/avdata99/ckan/commit/20d35a04bfa1074308928f2fec5e23bb4ef92c37) @amercader